src/test/java/org/apache/commons/compress/archivers/JarTestCase.java

```diff
@@ -61,6 +61,9 @@ public void testJarUnarchive() throws Exception {
 
         ZipArchiveEntry entry = (ZipArchiveEntry)in.getNextEntry();
         File o = new File(dir, entry.getName());
+        if (!o.toPath().normalize().startsWith(dir.toPath().normalize())) {
+            throw new RuntimeException("Bad zip entry");
```

Comment:
Don't throw RuntimeException, it's an anti-pattern; one of ISE or IAE is likely better depending on what's at fault.